carbon-design-system/carbon-for-ibm-dotcom
# fix(table-of-contents): prevent extra space above Table of Contents

### Related Ticket(s)

Resolves #5577 
HC Jira ticket: https://jsw.ibm.com/browse/HC-2152

### Description

A couple key things to keep in mind:
1. It looks like browser stylesheets give Web Component wrappers `display: inline` by default in some cases.
2. Elements with no specified height that are set to display inline and are given horizontal padding are given an implicit height equal to their `font-size`.

The `<dds-table-of-contents>` wrapper element was displaying inline in this case, and as such it was receiving an implicit height from the horizontal padding applied to it by the `.bx--grid` class. This was fixed by setting `<dds-table-of-contents>` to `display: block;`. 

This "activated" the wrapper element's horizontal padding rules, so I removed the now-unnecessary grid wrappers from the `Dotcom Shell` demo content. This prevents redundant padding and keeps all elements in the demos properly aligned.

#### Screenshots
(the Leadspace is given a temporary grid background to make the change easier to see)

- **Desktop**
   - Before:
   <img width="1915" alt="5577-toc-space--desktop--before" src="https://user-images.githubusercontent.com/12532727/130807905-8d8608dc-2086-4f86-a75a-33b009a5100c.png">

   - After: 
   <img width="1915" alt="5577-toc-space--desktop--after" src="https://user-images.githubusercontent.com/12532727/130807942-549a1617-073e-4d08-931d-f0ddb983b5a6.png">

- **Tablet**
   - Before: 
   <img width="959" alt="5577-toc-space--tablet--before" src="https://user-images.githubusercontent.com/12532727/130808103-f4b50fc9-ee96-4d05-9087-a8f17dafa55e.png">

   - After: 
   <img width="959" alt="5577-toc-space--tablet--after" src="https://user-images.githubusercontent.com/12532727/130808120-db77162c-cf76-4354-bd46-42e980d0fefc.png">
- **Mobile**
   - Before:
   <img width="656" alt="5577-toc-space--mobile--before" src="https://user-images.githubusercontent.com/12532727/130808160-2be1fbb6-c84e-40a8-9640-2fbe4cb9e385.png">

   - After:
   <img width="656" alt="5577-toc-space--mobile--after" src="https://user-images.githubusercontent.com/12532727/130808170-e39aac51-594e-46a2-9ac6-55276ae88042.png">

### Changelog
**Changed**

- Ensure `<dds-table-of-contents>` elements' grid padding rules are respected and no extra space appears above and/or below them.

**Removed**

- Removed unnecessary grid wrappers from dotcom-shell demo content.

### Testing Instructions
1. Visit both Web Components `Dotcom Shell` component variations that show the ToC in vertical and horizontal orientations:
   - Vertical: `?path=/story/components-dotcom-shell--default`
   - Horizontal: `?path=/story/components-dotcom-shell--with-horizontal-toc`
2. Verify the extra space above the ToC has been removed and no visual regressions have been introduced.
   - Though I did not find any in my testing, this PR's changes are most likely to cause an unwanted horizontal layout shift.
